File: src/main/java/dev/velvet/util/packet/PacketUtils.java
package dev.velvet.util.packet;

import net.minecraft.client.Minecraft;
import net.minecraft.network.Packet;
import net.minecraft.network.ThreadQuickExitException;

@SuppressWarnings("unused")
public class PacketUtils {
    //I hate to do it, but I had many issues with types with this class in Kotlin, so I just used Java instead where it works fine.

    public static void send(Packet packet) {
        Minecraft.getMinecraft().getNetHandler().addToSendQueue(packet);
    }
    public static void handle(Packet packet, boolean outgoing) {
        if (outgoing) {
            Minecraft.getMinecraft().getNetHandler().netManager.sendPacket(packet);
        } else {
            if (Minecraft.getMinecraft().getNetHandler().netManager.channel.isOpen()) {
                try {
                    packet.processPacket(Minecraft.getMinecraft().getNetHandler());
                }
                catch (final ThreadQuickExitException ex) {
                    throw new RuntimeException(ex);
                }
            }
        }
    }
}

File: src/main/kotlin/dev/velvet/util/packet/PacketUtils.kt
package dev.velvet.util.packet

import net.minecraft.client.Minecraft
import net.minecraft.network.INetHandler
import net.minecraft.network.Packet
import net.minecraft.network.ThreadQuickExitException
import java.lang.RuntimeException

private val net = Minecraft.getMinecraft().netHandler.netManager

@Suppress("unused")
object PacketUtils {
    fun send(packet: Packet<INetHandler>) {
        Minecraft.getMinecraft().netHandler.addToSendQueue(packet)
    }
    fun handle(packet: Packet<INetHandler>, outgoing: Boolean) {
        if (outgoing) {
            net.sendPacket(packet)
        } else {
            if (net.channel.isOpen) {
                try {
                    packet.processPacket(net.packetListener)
                } catch (ex: ThreadQuickExitException) {
                    throw RuntimeException(ex)
                }
            }
        }
    }
}